python/mlc_chat/compiler_pass/fuse_add_norm.py

```diff
@@ -0,0 +1,273 @@
+"""A compiler pass that fuses add + rms_norm."""
+import tvm
+from tvm import IRModule, relax
+from tvm.relax.dpl import PatternContext, rewrite_bindings
+from tvm.relax.dpl.pattern import is_op, wildcard
+from tvm.relax.expr_functor import PyExprMutator, mutator
+from tvm.script import tir as T
+
+
+def get_add_rmsnorm_tir(hidden_size: int, is_decode=True):
+    @T.prim_func(private=True)
```

Comment:
I suppose the fused add-rmsnorm operator could be expressed by TE and scheduled by Dlight easily

Reply:
No it's not that easy. The tricky point is that add_rmsnorm function has 2 outputs: both results of add and rms_norm, which makes compute_inline/compute_at/reverse_compute_at all fail in such case. I have to write scheduled TIR to work around.

Reply:
Got it. Thanks for the elaboration!

Reply:
What's the implication on performance? If the manual schedule can't generalize to all cases we can try supporting such pattern in cublas fusion

Reply:
I heard from @MasterJH5574 that cublas fusion cannot successfully fuse matmul and divide_add now, so I create this small pass to unblock our effort on mlc serve perf profiling. In the long term, surely this can be replaced by better cublas fusion, but it doesn't hurt to work as a fallback or as a target to compare.